uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -591,8 +601,21 @@ define([
 
         return loadStatuses.call(self)
           .then(function () {
-            initUserRole.call(self);
+            return initUserRole.call(self);
+          })
+          .then(function () {
             initOpenMode.call(self);
+
+            if (self.directiveOptions.forStaff) {
+              return loadManagees.call(self);
+            }
+          })
+          .then(function () {
+            if (self.directiveOptions.forStaff && !self.contact) {
```

Comment:
why don't you simply call _init() when the contact has been assigned, so you can avoid this branching